files/routes/volunteer.py

```diff
@@ -0,0 +1,66 @@
+
+from datetime import datetime, timedelta
+from files.__main__ import app
+from files.classes.user import User
+import files.helpers.jinja2
+from files.helpers.wrappers import auth_required
+from files.routes.volunteer_common import VolunteerDuty
+import files.routes.volunteer_janitor
+from flask import render_template, g, request
+from os import environ
+import sqlalchemy
+from typing import Optional
+import pprint
+
+
+
+
+@files.helpers.jinja2.template_function
+def volunteer_available_for(u: Optional[User]) -> bool:
+    return volunteer_get_duty(u) is not None
+
+def volunteer_get_duty(u: Optional[User]) -> Optional[VolunteerDuty]:
+    if u is None:
+        return None
+    
+    # check to make sure it's at least 20h since the last volunteer
+    if not app.config['DBG_VOLUNTEER_PERMISSIVE']:
```

Comment:
this doesn't seem to persist across reboots?

Reply:
Comment's in kinda the wrong place - fixed. The config is meant to be set via server config, mostly because that's easy for me :)